### PR TITLE
Gun Dual Wielding + Accuracy Tweaks

### PR DESCRIPTION
**Pull request by BurgerLUA**
# Overview
Other servers have this feature. If you fire a gun in one hand, you will fire the gun in the other hand, if any. Self-explanatory. Doesn't matter if they are one and the same.

All weapons can be held with two hands, including pistols. All weapons are less accurate when held with one weapon, with differing varieties based on the weapon's shape and weight.

# New Traitor Item
A "Tactical Harness" is added as a traitor item. Basically you can perform an easy reload by attacking the worn belt with a pistol, as long as you have a magazine that fits in it.

## Potential Issues
It was thoroughly tested. Aiming is fine. Shooting is fine. The only concern I believe is balance (cough, .357 revolvers, cough).

### CODE REVIEW

**Comment by BurgerLUA** [comment]
Also this contains 0 CM code.

File: code/modules/projectiles/gun.dm [hunk 1]
```diff
@@ -264,6 +266,16 @@
 	user.setMoveCooldown(move_delay)
 	next_fire_time = world.time + fire_delay
 
+	if(enable_dual_fire && dual_fire)
+		var/obj/item/weapon/gun/the_gun = user.get_inactive_hand()
+		if(istype(the_gun,/obj/item/weapon/gun/offhand) || !istype(the_gun) || !the_gun.enable_dual_fire)
+			return
+
+		spawn(fire_delay/2)
```
Comment (LordFowl) [inline code/modules/projectiles/gun.dm]: This could very easily be a timer.

**Comment by BurgerLUA** [comment]
I will make it so that every gun can be held by two hands. If the gun is held with two hands, then there are no accuracy penalties. If the gun isn't held by two hands, then there are accuracy penalties.

**Comment by Alberyk** [comment]
Or just make so that the gun being fired in the second hand has a penalty.

**Comment by BurgerLUA** [comment]
I think that would be a little arbitrary comapared to my suggestion. I'm making it so that weapons, depending on their type and size, are less accurate when held with one hand, and more accurate when held with two.

**Comment by BurgerLUA** [comment]
Okay, I gave almost every projectile/laser/whatever weapon an accuracy stat. For some reason, many of the weapons had a default accuracy stat despite it being a weapon that would likely have a different accuracy stat.

I will have to look through it again to see if I didn't fuck up anything.

**Comment by BurgerLUA** [comment]
!wip

**Comment by BurgerLUA** [comment]
I recommend an intense review of the current accuracy values of the weapons. I believe balance testing will likely be needed.

**Comment by BurgerLUA** [comment]
!review

**Comment by OneOneThreeEight** [comment]
What's the issue with just allowing people to alternate firing weapons in various  hands? X-Click-X-Click-X-Click.

Combat is not really a focus of this server either, so I find this to be an unnecessary change. It's insidious powercreep.

**Comment by BurgerLUA** [comment]
Er if you read the PR, there is a serious accuracy debuff from dual wielding.

**Comment by LordFowl** [comment]
@BurgerLUA The feedback required tag was added 5 days ago. Please create a forum thread so these balance discussions can be better hosted.

**Comment by BurgerLUA** [comment]
@LordFowl I am currently working an additions that involve accuracy and at this point it would be premature to create a project thread.

**Comment by HeyBanditoz** [comment]
@OneOneThreeEight PRs should generally be left to code reviews, not suggestions. Take it to the future forum thread if you feel the need.

**Comment by BurgerLUA** [comment]
Reopening this. The code here is good and done and there's no point in closing this.

**Comment by BurgerLUA** [comment]
Since that was a pretty big upstream merge, I'll have to test it a few times to see if it still works.

**Comment by BurgerLUA** [comment]
Everything looks good, I think. Works still.

**Comment by BurgerLUA** [comment]
Outdated PR that needs to be readded.

**Pull request closed without merging**